app/background/node/service.js

```diff
@@ -436,10 +482,50 @@ export class NodeService extends EventEmitter {
   }
 
   async getCoin(hash, index) {
+    if (await this.getSpvMode()) {
+      return hapiGet(`/coin/${hash}/${index}`);
+    }
     return this.client.getCoin(hash, index);
   }
 
-  async verifyMessageWithName(name, signature, message) {
+  async verifyMessageWithName(name, signature, str) {
+    if (await this.getSpvMode()) {
+      const result = await this.getNameInfo(name);
+      const owner = result?.info?.owner;
+
+      if (!owner) {
+        throw new Error('Cannot find the name owner.');
+      }
+
+      const coin = await this.getCoin(owner.hash, owner.index);
+
+      if (!coin) {
+        throw new Error('Cannot find the owner\'s address.');
+      }
+
+      const addr = Address.fromString(coin.address, this.network);
+
+      if (addr.version !== 0 || addr.hash.length !== 20)
+        return false;
+
+      const MAGIC_STRING = `${pkg.currency} signed message:\n`;
```

Comment:
The thing I want to avoid tho is sending the signature to hosted node. It seems better to be able to verify signature locally, instead of relying on a hosted node to tell me. Perhaps we can make it an exportable util function in hsd?

Reply:
SPV nodes still have rpc verifymessage, it just takes an addres as input instead of the name. So once you get the corresponding address from the API you can just call that locally. 

Also, why not send the signature to the API server?

Reply:
Trust but verify is the general model. We are trusting the hosted node for things that we could verify locally.

For verifymessage, doing it locally means that we only need to trust the node to send us the correct name owner (which could be independently verify) and then verify signature locally. If we rely on the node, we would only get a boolean of whether or not the message is valid, but no ways to verify independently.

Reply:
> SPV nodes still have rpc verifymessage

Ah, I will do this

Reply:
actually - even trusting the full node to return the owner coin should technically be provable - using Urkel. I don't think we have this in hsd quite yet, but an SPV node could use the same `pool.lookup` function it uses to resolve TLDs with Urkel Proofs, but instead of returning the dns resource, it could return the owner. (The entire `Namestate` object is included in urkel, including `owner`).